examples/chip-tool/commands/common/Command.cpp

```diff
@@ -91,10 +95,66 @@ bool Command::InitArgument(size_t argIndex, const char * argValue)
         break;
     }
 
-    case ArgumentType::String: {
+    case ArgumentType::CharString: {
         const char ** value = reinterpret_cast<const char **>(arg.value);
-        *value              = const_cast<char *>(argValue);
-        isValidArgument     = (strcmp(argValue, *value) == 0);
+        *value              = argValue;
+        isValidArgument     = true;
+        break;
+    }
+
+    case ArgumentType::OctetString: {
+        auto * value = static_cast<chip::ByteSpan *>(arg.value);
+        // We support two ways to pass an octet string argument.  If it happens
+        // to be all-ASCII, you can just pass it in.  Otherwise you can pass in
+        // 0x followed by the hex-encoded bytes.
+        size_t argLen                     = strlen(argValue);
+        static constexpr char hexPrefix[] = "hex:";
+        constexpr size_t prefixLen        = ArraySize(hexPrefix) - 1; // Don't count the null
+        if (strncmp(argValue, hexPrefix, prefixLen) == 0)
+        {
+            // Hex-encoded.  Decode it into a temporary buffer first, so if we
+            // run into errors we can do correct "argument is not valid" logging
+            // that actually shows the value that was passed in.  After we
+            // determine it's valid, modify the passed-in value to hold the
+            // right bytes, so we don't need to worry about allocating storage
+            // for this somewhere else.  This works because the hex
+            // representation is always longer than the octet string it encodes,
+            // so we have enough space in argValue for the decoded version.
+            auto buffer = static_cast<uint8_t *>(chip::Platform::MemoryCalloc(argLen, 1)); // Bigger than needed, but it's fine.
```

Comment:
Could we use ScopedBuffer.h to allocate and free automatically?

Reply:
Good catch, will do.